### PR TITLE
docker: Allow deleting containers from lists

### LINKED ISSUE #1517: Should be able to multi-select containers/images for deletion
When testing with docker containers, it's not uncommon to build up a large number of unused containers. It would be very handy to be able to select a multitude of them and delete them all (or start or stop them all, etc.)

### PR DESCRIPTION
**Pull request by petervo**
Fixes #1517

### CODE REVIEW

File: pkg/shell/cockpit-docker.js [hunk 1]
```diff
@@ -370,6 +370,24 @@ function render_container (client, $panel, filter_button, prefix, id, container)
     var added = false;
     if (!tr.length) {
         var img_waiting = $('<div class="waiting">');
+        var btn_delete = $('<button class="btn btn-danger pficon pficon-close btn-delete">').
+            on("click", function() {
+                var self = this;
+                $(this).hide().
```
Comment (stefwalter) [inline pkg/shell/cockpit-docker.js]: This should probably say 'self' for consistency.

File: pkg/shell/cockpit-docker.js [hunk 2]
```diff
@@ -370,6 +370,24 @@ function render_container (client, $panel, filter_button, prefix, id, container)
...
+                    siblings("div.waiting").show();
+                client.rm(id, true).
+                    fail(function(ex) {
+                        shell.show_unexpected_error(ex);
+                        $(self).show().
+                            siblings("div.waiting").show();
```
Comment (stefwalter) [inline pkg/shell/cockpit-docker.js]: Why do you show sublings of div.waiting above, *and* here? Is that correct?

File: pkg/shell/cockpit-docker.js [hunk 3]
```diff
@@ -370,6 +370,24 @@ function render_container (client, $panel, filter_button, prefix, id, container)
...
+                    });
+                return false;
+            });
+        var wrapped = $('<div class="button-wrap-tooltip">').
```
Comment (stefwalter) [inline pkg/shell/cockpit-docker.js]: Could you add a comment about why this is necessary? About why the tooltip isn't added directly to the button?

**Comment by stefwalter** [comment]
Nice. This does make it much easier to delete containers. Some more design comments after trying this out:

 * When you go do something else (view container details) when in danger mode, danger mode should be disabled. It should not be active when you come back.
 * I don't think we should show the [v] check box when there are no containers to select. It just raises too many questions, in my opinion. Same applies on the images page, when no containers of that page are running.
 * How does one know that the [x] means delete? The exact same icon means close in other places in Patternfly, for example in a dialog header. Should we use a different icon, or at a minimum put a tooltip: "Delete immediately"?

**Comment by andreasn** [comment]
There is a specific icon in Patternfly for delete specifically. pficon-delete https://www.patternfly.org/visual-styles/icons-glyphs/

screenshot from 2015-01-27 11 27 18

Renders kind of ugly here though. :/
For that reason I'm leaning towards the x button with a tooltip. I feel the red is a quite strong indicator as well. This could be enhanced with the Delete button in the container details becoming red as well (I can do that as a separate pull request)

The tooltip "you can only delete stopped containers" needs to start with an uppercase.

**Comment by andreasn** [comment]
Managed to work around the crap rendering by increasing the of the font-size a bit to 14px, so in that case it should be ok to go with pficon-delete.

screenshot from 2015-01-27 11 46 18

**Comment by andreasn** [comment]
To keep consistency within cockpit, we also need to change the icon for removing of hosts on the dashboard and for removing rows in the Run Image dialog.

**Comment by petervo** [comment]
Can changing the delete icons everywhere be a separate issue/PR?

**Comment by stefwalter** [comment]
> Can changing the delete icons everywhere be a separate issue/PR?

Yup. Can you file it?

**Comment by petervo** [comment]
More review needed, the disabled global handler is needed for other UI work as well so if this will site awhile maybe we should pull that into a different PR. LMK

Leaving the icon changes to #1735

**Comment by stefwalter** [comment]
Nice. Checked that spacebar and enter also don't activate the disabled buttons.

**Comment by stefwalter** [comment]
Nice. Squashed and merged.

**Pull request closed without merging**